lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Task/TaskClientCodeException.cs

```diff
@@ -16,15 +16,41 @@
 // under the License.
 
 using System;
-using Org.Apache.REEF.Tang.Interface;
+using System.Runtime.Serialization;
+using System.Security.Permissions;
 
 namespace Org.Apache.REEF.Common.Runtime.Evaluator.Task
 {
-    internal sealed class TaskClientCodeException : Exception
+    /// <summary>
+    /// An Exception thrown when Task operations (Start, Stop, Suspend) fail.
+    /// </summary>
+    [Serializable]
+    public sealed class TaskClientCodeException : Exception
```

Comment:
This will be exposed to the users for them to differentiate a control flow error vs an actual Task execution error, as is done on the Java side.